aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/TransferWorkerObserver.kt

```diff
@@ -186,8 +186,14 @@ internal class TransferWorkerObserver private constructor(
                 )
             )?.use {
                 while (it.moveToNext()) {
-                    val id = it.getInt(it.getColumnIndexOrThrow(TransferTable.COLUMN_ID))
-                    attachObserver(id.toString())
+                    // getInt has been reported to  throw in rare cases where it appears another thread
+                    // may have closed cursor
+                    try {
+                        val id = it.getInt(it.getColumnIndexOrThrow(TransferTable.COLUMN_ID))
+                        attachObserver(id.toString())
+                    } catch (exception: IllegalStateException) {
+                        // do nothing. Cursor is likely closed
+                    }
```

Comment:
https://developer.android.com/reference/android/database/Cursor#getColumnIndexOrThrow(java.lang.String)

IllegalStateException is the only exception type to be handled here. Since these Android APIs are written in Java, we can trust that the exceptions thrown should only be the ones listed in documentation.